vllm/core/block_manager.py

```diff
@@ -184,7 +179,8 @@ def append_slot(self, seq: Sequence) -> Optional[Tuple[int, int]]:
             else:
                 # The sequence has a new logical block.
                 # Allocate a new physical block.
-                block = self.gpu_allocator.allocate()
+                block = self.gpu_allocator.allocate(
```

Comment:
`BlockSpaceManager.append_slot` is used to allocate memory in the decode case. 2 high level scenarios:
- There is space in the current physical block, so we use that
- A new block is allocated

Since the actual tokens in the block will change on decode, I think we need to be updating the hash of the physical each iteration in all cases (right now we only update when new blocks are created).

One complexity is that we are allocating space right BEFORE the new token is generated in the current `engine_step`. This means that the hash will be immediately invalidated. This will cause issues because if a sequence ends after the token is generated, we will never get to update the hash and there will be an invalid hash. Perhaps an approach to take is that we only hash/cache "full" blocks, which would remove this issue (I am not sure how we get around it otherwise)




As a result, in this step, we need to be updating the hash of the physical block in all cases. Currently, we are only creating 



- B) the current block is full, in which case we allocate a new block